### DISCUSSION TITLE
Increasing the number of collections from 65K to Unlimited.

### DISCUSSION THREAD

**Discussion by SupreethRao99** [milvus-io/milvus#29609] [Q&A and General discussion] [answered]
Hi,

The Milvus limits page (https://milvus.io/docs/limitations.md) says that a maximum of 65K collections can be stored on Milvus, is there any way this can be increased? 
An example scenario where this could be an issue is let's say allocation of a new collection to a user. won't this limit the number of users to 65K? 
Am I thinking of this wrong, could anyone suggest a better way to handle this?

**Comment by xiaofan-luan** [comment]
We suggest to use partitionkey features rather than use collection per tenant.
Currently we seen great performance issue with 10000+ collections

**Reply by SupreethRao99** [reply]
would this be a good starting point (https://milvus.io/docs/multi_tenancy.md) to learning more about it? thank you!

**Reply by yhmo** [reply, marked as answer]
> would this be a good starting point (https://milvus.io/docs/multi_tenancy.md) to learning more about it? thank you!

Yes, read the doc to learn more.

Milvus uses time synchronization mechanism to keep data consistency: https://milvus.io/docs/time_sync.md#Time-synchronization-system-timetick
For each collection, the server requires some resources to observe the collection and maintain data consistency. So, with a large number of collections, you will see very high CPU usage and the server runs into slow response.

Partition requires fewer resources, this is why we recommend using partitions rather than collections.
